test/extensions/filters/http/jwt_authn/filter_test.cc

```diff
@@ -288,8 +296,139 @@ TEST_F(FilterTest, OutBoundForbiddenFailure) {
+  EXPECT_EQ(1U, mock_config_->stats().allowed_.value());
+
+  Buffer::OwnedImpl data("");
+  EXPECT_EQ(Http::FilterDataStatus::Continue, filter_->decodeData(data, false));
+  EXPECT_EQ(Http::FilterTrailersStatus::Continue, filter_->decodeTrailers(trailers_));
+}
+
+// Test bypass requirement from per-route config
+TEST_F(FilterTest, TestPerRouteBypass) {
+  EXPECT_CALL(filter_callbacks_, route()).WillOnce(Return(mock_route_));
+  EXPECT_CALL(mock_route_->route_entry_, perFilterConfig(HttpFilterNames::get().JwtAuthn))
+      .WillOnce(Return(per_route_config_.get()));
+
+  // findVerifier is not called.
+  EXPECT_CALL(*mock_config_.get(), findVerifier(_, _)).Times(0);
+  // If findPerRouteVerifier is called, and return nullptr, it means bypass
+  EXPECT_CALL(*mock_config_.get(), findPerRouteVerifier(_))
+      .WillOnce(Return(std::make_pair(nullptr, EMPTY_STRING)));
+
+  auto headers = Http::TestRequestHeaderMapImpl{};
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(headers, false));
+  EXPECT_EQ(1U, mock_config_->stats().allowed_.value());
+
+  Buffer::OwnedImpl data("");
+  EXPECT_EQ(Http::FilterDataStatus::Continue, filter_->decodeData(data, false));
+  EXPECT_EQ(Http::FilterTrailersStatus::Continue, filter_->decodeTrailers(trailers_));
+}
+
+// Test per-route config with wrong requirement_name
+TEST_F(FilterTest, TestPerRouteWrongRequirementName) {
+  EXPECT_CALL(filter_callbacks_, route()).WillOnce(Return(mock_route_));
+  EXPECT_CALL(mock_route_->route_entry_, perFilterConfig(HttpFilterNames::get().JwtAuthn))
+      .WillOnce(Return(per_route_config_.get()));
+
+  // findVerifier is not called.
+  EXPECT_CALL(*mock_config_.get(), findVerifier(_, _)).Times(0);
+  // If findPerRouteVerifier is called, and return error message.
+  EXPECT_CALL(*mock_config_.get(), findPerRouteVerifier(_))
+      .WillOnce(
+          Return(std::make_pair(nullptr, "Wrong requirement_name: abc. Correct names are: r1,r2")));
+
+  auto headers = Http::TestRequestHeaderMapImpl{};
+  EXPECT_EQ(Http::FilterHeadersStatus::StopIteration, filter_->decodeHeaders(headers, false));
+  EXPECT_EQ(1U, mock_config_->stats().denied_.value());
+
+  Buffer::OwnedImpl data("");
+  EXPECT_EQ(Http::FilterDataStatus::Continue, filter_->decodeData(data, false));
+  EXPECT_EQ(Http::FilterTrailersStatus::Continue, filter_->decodeTrailers(trailers_));
+  EXPECT_EQ(filter_callbacks_.details(),
+            "jwt_authn_access_denied{Wrong requirement_name: abc. Correct names are: r1,r2}");
+}
+
+// Test verifier from per-route config
+TEST_F(FilterTest, TestPerRouteVerifierOK) {
+  EXPECT_CALL(filter_callbacks_, route()).WillOnce(Return(mock_route_));
+  EXPECT_CALL(mock_route_->route_entry_, perFilterConfig(HttpFilterNames::get().JwtAuthn))
+      .WillOnce(Return(per_route_config_.get()));
+
+  // findVerifier is not called.
+  EXPECT_CALL(*mock_config_.get(), findVerifier(_, _)).Times(0);
+  // If findPerRouteVerifier is called
```

Comment:
Seems missing the second half of the comment? `If findPerRouteVerifier is called, and...`

Reply:
Done